go/vt/vtgate/vtgate.go

```diff
@@ -245,6 +247,12 @@ func (vtg *VTGate) ExecuteKeyspaceIds(ctx context.Context, sql string, bindVaria
 		return errTooManyInFlight
 	}
 
+	if len(keyspaceIds) == 1 {
```

Comment:
I think this if statement can be pushed inside AnnotateKeyspaceIdIfDML. Would be more robust.